src/FunderMaps.Infrastructure/Storage/SpacesBlobStorageService.cs

```diff
@@ -189,26 +194,109 @@ public async Task StoreDirectoryAsync(string directoryName, string directoryPath
         {
             try
             {
-                async Task DirectorySearch(string path, string subdir = "")
+                TransferUtilityUploadDirectoryRequest request = new()
+                {
+                    BucketName = _options.BlobStorageName,
+                    Directory = directoryPath,
+                    KeyPrefix = directoryName,
+                    SearchOption = SearchOption.AllDirectories,
+                    UploadFilesConcurrently = true,
+                };
+
+                request.UploadDirectoryFileRequestEvent += (sender, e) =>
                 {
-                    foreach (var file in Directory.GetFiles(path))
+                    e.UploadRequest.CannedACL = (storageObject?.IsPublic ?? false) ? S3CannedACL.PublicRead : S3CannedACL.Private;
+                    e.UploadRequest.Headers.ContentType = storageObject?.ContentType ?? e.UploadRequest.Headers.ContentType;
+                    e.UploadRequest.Headers.CacheControl = storageObject?.CacheControl ?? e.UploadRequest.Headers.CacheControl;
+                    e.UploadRequest.Headers.ContentDisposition = storageObject?.ContentDisposition ?? e.UploadRequest.Headers.ContentDisposition;
+                    e.UploadRequest.Headers.ContentEncoding = storageObject?.ContentEncoding ?? e.UploadRequest.Headers.ContentEncoding;
+                };
+
+                TransferUtilityConfig config = new()
+                {
+                    /** Note: This is currently set to the default value of 10. I did some benchmarking on 23 dec 2020 
+                           and discovered that turning this value up will cause some unstable behaviour resulting in
+                           the task throwing an exception and being cancelled. Setting this to 20 seemed to work fine, 
+                           however 50 or above seems to result in crashes. I've left it on 10 for now to be safe for use 
+                           in production. Worth investigating later for a major performance increase!
+
+                           - Patrick
+                    **/
+                    ConcurrentServiceRequests = ConcurrentServiceRequests
+                };
+
+                await new TransferUtility(client, config).UploadDirectoryAsync(request);
+            }
+            catch (AmazonS3Exception e)
+            {
+                _logger.LogError(e, "Could not store directory to Spaces using S3");
```

Comment:
```suggestion
                _logger.LogError("Could not store directory to Spaces using S3");
```

No need for stacktrace dump. The exception is captured in the next line.